app/src/org/odk/collect/android/views/ODKView.java

```diff
@@ -197,11 +197,17 @@ public ODKView(Context context, FormEntryPrompt[] questionPrompts, FormEntryCapt
     }
     
     public void removeQuestionFromIndex(int i){
-        mView.removeView((View) widgets.get(i));
         int dividerIndex = Math.max(i - 1, 0);
-        mView.removeView(dividers.get(dividerIndex));
-        widgets.remove(i);
-        dividers.remove(dividerIndex);
+
+        if (dividerIndex < dividers.size()) {
+            mView.removeView(dividers.get(dividerIndex));
+            dividers.remove(dividerIndex);
+        }
+
+        if (i < widgets.size()) {
+            mView.removeView((View) widgets.get(i));
+            widgets.remove(i);
+        }
```

Comment:
OOC what situation is causing this to be called with an index that doesn't exist? Is there another bug further upstream?